package.json

```diff
@@ -111,7 +112,8 @@
       "stage-1"
     ],
     "plugins": [
-      "transform-decorators-legacy"
+      "transform-decorators-legacy",
+      ["transform-runtime", { "polyfill": true }]
```

Comment:
Good catch.  I went and read thoroughly the docs for the runtime plugin - https://babeljs.io/docs/plugins/transform-runtime/ .  Not only does this automatically include the polyfill which replaces the promise polyfill it also optimizes the compiled code if I'm reading correctly.

The only danger as mentioned in the docs is that Map, Set, and Promise are globals, and as such this would not be good in a library such that it would affect a larger application.  I think we're ok implementing here.